src/rules/mergeableNamespaceRule.ts

```diff
@@ -0,0 +1,54 @@
+/**
+ * @license
+ * Copyright 2016 Palantir Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as ts from "typescript";
+import * as Lint from "../lint";
+
+export class Rule extends Lint.Rules.AbstractRule {
+    public static FAILURE_STRING = "mergeable namespace: ";
+
+    public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
+        const languageService = Lint.createLanguageService(sourceFile.fileName, sourceFile.getFullText());
+        const mergeableNamespaceWalker = new MergeableNamespaceWalker(sourceFile, this.getOptions(), languageService);
+        return this.applyWithWalker(mergeableNamespaceWalker);
+    }
+}
+
+class MergeableNamespaceWalker extends Lint.RuleWalker {
+   private languageService: ts.LanguageService;
```

Comment:
indentation looks slightly off here -- can you make sure every line uses 4 spaces?